ui/src/app/shared/services/applications-service.ts

```diff
@@ -85,30 +85,38 @@ export class ApplicationsService {
             .then(res => res.body as models.ManifestResponse);
     }
 
-    public updateSpec(appName: string, spec: models.ApplicationSpec): Promise<models.ApplicationSpec> {
+    public async updateSpec(appName: string, spec: models.ApplicationSpec): Promise<models.ApplicationSpec> {
+        const csrfToken = await requests.getCsrfToken();
```

Comment:
I think it should be possible to encapsulate token fetching in [requests.ts](https://github.com/argoproj/argo-cd/blob/master/ui/src/app/shared/services/requests.ts) and avoid changing each service method. Trying to implement it. I will update you once I find the way to do it.